### ISSUE TITLE
Support retuning of log level on spire-agent

### ISSUE THREAD

**Issue by edwbuck** [spiffe/spire#4986]
The previous issue covered server and agent tuning, and was rebased to server-only and closed successfully, this issue covers brining the agent up to the same log level tuning parity as the server.

Post-launch, it would be beneficial to adjust the SPIRE agent log level. The agent should read the initial log level from the config file path / command line options but then accept a cli command to adjust the SPIRE Server instance previously launched with `spire-server run ...`

From prior discussions:

* This will occur from a cli command on the agent "admin" UDS.
* This will use the same cli commands as the current spire-server implementation.

Work for the supporting API is already in place in spire-api-sdk, the Agent simply needs to implement and service the API.

**Comment by azdagron** [comment]
Some of the text in this description are referencing the server? This is for the agent, correct?

**Comment by edwbuck** [comment]
Yes, this is for the agent.  I'll update the items that didn't get edited as I copied from the original effort, which did reference the server.

**Comment by azdagron** [comment]
Great, thanks.

**Comment by faali1** [comment]
+1 to this :) Would be great to be able to adjust the spire-agent log level on the fly

**Comment by szvincze** [comment]
> +1 to this :) Would be great to be able to adjust the spire-agent log level on the fly

+1   same wish from my side.

**Comment by alitta-german** [comment]
+1 it would be great

**Comment by sorindumitru** [comment]
This is implemented now
